File: WebExAutoRecord/InputForm.vb
Imports System.Threading
Public Class InputForm
    Private ReadOnly WaitEvent As New ManualResetEvent(False)
    Private Sub OK_Click(sender As Object, e As EventArgs) Handles OK.Click
        If InputText.Text Is Nothing Or InputText.Text = "" Then
            MessageBox.Show("Please input something.")
        Else
            Me.Close()
        End If
    End Sub

    Public Function AskForInput(Query As String)
        Label1.Text = Query
        If Query.Contains("password") Then
            InputText.PasswordChar = "*"
        Else
            InputText.PasswordChar = Nothing
        End If
        InputText.Text = ""
        Dim OldHeight As Integer = InputText.Location.Y


        If Query.Contains("video:") Then

            Dim p As Point = InputText.Location

            p.Y = p.Y + InputText.Height + 5

            InputText.Location = p
            p.X = OK.Location.X
            OK.Location = p

            Skip.Visible = True
            Me.Height = Me.Height + InputText.Height + 10
        End If

        Do While InputText.Text = "" Or InputText.Text Is Nothing
            Me.ShowDialog()
        Loop

        If Query.Contains("video:") Then
            Skip.Visible = False
            Dim p As Point
            p.X = InputText.Location.X
            p.Y = OldHeight
            InputText.Location = p
            p.X = OK.Location.X
            OK.Location = p
            Me.Height = Me.Height - InputText.Height - 10
        End If

        Return InputText.Text
    End Function

    Private Sub Skip_Click(sender As Object, e As EventArgs) Handles Skip.Click
        InputText.Text = "0"
        Me.Close()
    End Sub

    Private Sub InputForm_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        If StartupForm.IsItalian Then
            Skip.Text = "Salta"
        End If
    End Sub
End Class

File: WebExAutoRecord/Forms/InputForm.cs
using System;
using System.Drawing;
using System.Windows.Forms;

namespace PoliDLGUI.Forms
{
    public partial class InputForm
    {
        public InputForm(Point location)
        {
            this.Location = location;
            InitializeComponent();
            _OK.Name = "OK";
            _Skip.Name = "Skip";
        }

        //private readonly ManualResetEvent WaitEvent = new ManualResetEvent(false);

        private void OK_Click(object sender, EventArgs e)
        {
            if (InputText.Text is null | string.IsNullOrEmpty(InputText.Text))
            {
                MessageBox.Show("Please input something.");
            }
            else
            {
                Close();
            }
        }

        public static object AskForInput(string Query, Point location)
        {
            InputForm inputForm = new InputForm(location);
            return inputForm.AskForInputInternal(Query);
        }

        public object AskForInputInternal(string Query)
        {
            //Should I fix this so it uses anchors instead of this weird position jank? Yes.
            //Am I going to? No.

            Label1.Text = Query;
            if (Query.Contains("password") & !Query.Contains("video:"))
                InputText.PasswordChar = '*';
            else
                InputText.PasswordChar = default;

            InputText.Text = "";
            int OldHeight = InputText.Location.Y;
            if (Query.Contains("video:"))
            {
                var p = InputText.Location;
                p.Y = p.Y + InputText.Height + 5;
                InputText.Location = p;
                p.Y += 20;
                p.X = OK.Location.X;
                OK.Location = p;
                Skip.Visible = true;
                Height = Height + InputText.Height + 10 + 20;
                reUsePW.Visible = true;
            }

            while (string.IsNullOrEmpty(InputText.Text) | InputText.Text is null)
            {
                ShowDialog();
                Activate();
            }
               
            if (Query.Contains("video:"))
            {
                Skip.Visible = false;
                var p = default(Point);
                p.X = InputText.Location.X;
                p.Y = OldHeight;
                InputText.Location = p;
                p.X = OK.Location.X;
                OK.Location = p;
                Height = Height - InputText.Height - 10 - 20;
            }
            if (!reUsePW.Checked)
                return InputText.Text;
            else
                return InputText.Text + "_ReUseForAllVideos!";  //Is this jank? 
        }

        private void Skip_Click(object sender, EventArgs e)
        {
            InputText.Text = "0";
            Close();
        }

        private void InputForm_Load(object sender, EventArgs e)
        {
            if (StartupForm.IsItalian)
            {
                Skip.Text = "Salta";
                reUsePW.Text = "Riutilizza password?";
            }
        }

        private void reUsePW_CheckedChanged(object sender, EventArgs e)
        {

        }
    }
}